examples/camera-app/linux/include/camera-device.h

```diff
@@ -33,12 +33,29 @@
 #define VIDEO_STREAM_GST_DEST_PORT 5000
 #define AUDIO_STREAM_GST_DEST_PORT 5001
 
-#define MAX_CONTENT_BUFFER_SIZE_BYTES (1024)
-#define MAX_ENCODED_PIXEL_RATE (10000)
-#define MAX_CONCURRENT_VIDEO_ENCODERS (1)
-#define MAX_NETWORK_BANDWIDTH_MBPS (64)
-#define MICROPHONE_MIN_LEVEL (1)
-#define MICROPHONE_MAX_LEVEL (254)
+// Camera Constraints set to typical values.
+// TODO: Look into ways to fetch from hardware, if required/possible.
+static constexpr uint32_t kMaxContentBufferSizeBytes = 4096;
+static constexpr uint32_t kMaxNetworkBandwidthMbps   = 128;
+static constexpr uint8_t kMaxConcurrentEncoders      = 1;
+static constexpr uint32_t kMaxEncodedPixelRate       = 27648000; // 720p at 30fps
+static constexpr uint8_t kMicrophoneMinLevel         = 1;
+static constexpr uint8_t kMicrophoneMaxLevel         = 254;  // Spec constraint
+static constexpr uint8_t kMicrophoneMaxChannelCount  = 8;    // Spec Constraint in AudioStreamAllocate
+static constexpr uint16_t kMinResolutionWidth        = 320;  // Low SD resolution
+static constexpr uint16_t kMinResolutionHeight       = 240;  // Low SD resolution
+static constexpr uint16_t kMaxResolutionWidth        = 1920; // 1080p resolution
+static constexpr uint16_t kMaxResolutionHeight       = 1080; // 1080p resolution
+static constexpr uint16_t kSnapshotStreamFrameRate   = 30;
+static constexpr uint16_t kMaxVideoFrameRate         = 120;
+static constexpr uint16_t kMinVideoFrameRate         = 15;
+static constexpr uint32_t kMinBitRateBps             = 10000;   // 10 kbps
+static constexpr uint32_t kMaxBitRateBps             = 2000000; // 2 mbps
+static constexpr uint32_t kMinFragLenMsec            = 1000;    // 1 sec
+static constexpr uint32_t kMaxFragLenMsec            = 10000;   // 10 sec
```

Comment:
Wondering how we can have a sensor width that is 4:3 AR 480p and a max resolution that is 16:9 AR 1080p.

Reply:
While technically, the sensor could have a different aspect ratio than the output resolution, changing it to be consistent 16:9.